### ISSUE TITLE
[FF] [Dev Build] Video Briefly Freezes at the Start while Audio Keeps Playing

### ISSUE THREAD

**Issue by beefcake3000** [YePpHa/YouTubeCenter#2007]
Upon first playing any video, it will play briefly (~1 second), freeze for ~2-3 seconds with the audio still playing as normal, and then resume playing. While it is frozen, the buffering circle on the video appears. 

If I disable YTC, the problem goes away. I am using YTC Dev Build 532, Firefox 40.0, HTML player, and have auto-buffering and auto-play disabled.

**Comment by Corruptinglyneedful** [comment]
I too am getting this issue.  It's exactly as you describe it as well.  It also causes the video to desync.

Is it because we disabled auto buffering?

**Comment by beefcake3000** [comment]
I don't think so. I disabled it after I realised that auto buffering disables the video thumbnail and I think I started having this issue afterwards. It makes sense that if auto buffering is enabled that the video wont need to freeze and buffer mid-way. 

Might also be because of Firefox version 40?

**Comment by Corruptinglyneedful** [comment]
No, I am on Firefox 39.0.3

**Comment by beefcake3000** [comment]
must have been after the most recent update then. I don't remember which version I started using YTC, but this certainly wasn't an issue when I first started. I think it was the version right before this one that worked.

**Comment by Corruptinglyneedful** [comment]
I think the issue has been present either 1 or 2 updates ago, yes.

Fix one thing, break another.  Heh.

**Comment by Corruptinglyneedful** [comment]
I think I found a work-around.

I disabled "Prevent auto-buffering" and it stopped doing it for me.  So you should try that.

**Comment by beefcake3000** [comment]
Yeah, but auto-buffering makes the thumbnails go away on the big player. They're useful to me because of the way I use YouTube: I open my subscriptions feed and open each video that interests me in a new tab. Sometimes I don't get to watch all of the videos that day, so videos can pile up. When I go through the tabs it's much easier to look quickly look at the large thumbnail (since most channels have custom, informative thumbnails) than read the video description. Also, since I have many videos open at once, the auto-buffering will take up my internet bandwidth, despite it being fairly large, and cause the video first video I watch to be slow to buffer. I also find that when many videos are buffered at once, FF bogs down a bit.

If my internet speed is 50Mbits down, I shouldn't need to pre-buffer a video to watch it. Why does the video work fine without the plugin and without pre-buffering?

**Comment by cwsault** [comment]
I noticed this as well. Narrowed it down to having autoplay + autobuffer disabled (which I also use for the thumbnails) along with DASH playback allowed. Disabling DASH playback seems to resolve the issue, but of course prevents playing in 1080p.

It seems to occur on most, but not all videos, but for videos on which it is present, it seems to happen consistently. No idea if related to the video itself or how DASH playback sends data to the user though. Example of a video I find it occuring consistently on: [Video](https://www.youtube.com/watch?v=p9PAuWV-Vn0)

I've noticed three ways this manifests -- one is as you described where the video freezes with the audio continuing, another the video will go black rather than freezing. Rarely, the video will continue normally, but with a noticable disturbance or pop in the audio (which can also be present when the video does freeze).

Quickly pulling the progress bar back to the starting point after playing the video seems to avoid this. Dunno if it's just because the that provides enough time for the first chunk to buffer, or if something else is at play. But I also noticed if I load the video with the time variable set to several seconds, it seems to reduce the freeze time. I'm getting less / zero occurences if the time value is set high enough.

It's worth noting that the userscript "YouTube Auto Buffer and Auto HD" seems able to prevent autoplay while usually avoiding this problem, when "Stop Loading Immediately" is selected. But whatever it's doing does seem to take longer to load up the actual video. Current version of the script drops this option, but a here is a link to and older beta version that seems to work: [[Author's Dropbox]](https://www.dropbox.com/s/35f6h6t9vtn9y4v/YT_AutoBuffer_1.2.88_beta.user.js?dl=0) [[Pastebin]](http://pastebin.com/iFSAHGhz) [[Archived Pastebin]](https://archive.is/WT0WG)

If the solution used there is unable to be sped up, several possible alternative approaches:
* When autoplay is disabled, detect the play event and briefly pause the video (assumes the player needs time to load the first block; could be toggleable)
* When autoplay is disabled and no time-variable is supplied, reset time to 0 on playing (assumes that resetting the progress bar has some other effect than simply providing time to load the first block)
* Obtain and overlay the thumbnail when pausing the video with auto-play disabled and auto-buffer enabled (probably a nice thing to have anyway)

**Comment by Corruptinglyneedful** [comment]
That is definitely something for Yeppa to look at.

**Comment by beefcake3000** [comment]
Indeed it is. Fixed the other issue I have with the page loading, but I still have this issue after all this time and right now what I'm doing is moving the time of the video back to start as soon as it plays (as @cwsault mentioned.)

**Comment by mpred04** [comment]
same here

**Comment by GlassedSilver** [comment]
This happens to me as well on Version 9.0 (11601.1.56) of Safari.
(OS X El Capitan, but this happened before as well)
